indico/modules/events/api.py

```diff
@@ -492,6 +457,41 @@ def category(self, idlist):
         query = self._update_query(query)
         return self.serialize_events(x for x in query if self._filter_event(x) and x.can_access(self.user))
 
+    def category_extra(self, ids):
+        has_future_events = (False if not self._toDT else
+                             True in [IndexesHolder().getById('categoryDateAll').hasObjectsAfter(cat_id, self._toDT)
+                                      for cat_id in self._idList])
+        return {
+            'eventCategories': self._build_category_path_data(ids),
+            'moreFutureEvents': has_future_events
+        }
+
+    def _build_category_path_data(self, ids):
+        return [{'_type': 'CategoryPath', 'categoryId': id, 'path': self.serialize_category_path(id)} for id in ids]
+
+    def serialize_category_path(self, id):
+        path = []
+        first_category = category = Category.get(id)
+        visibility_title = None
+        while category:
+            path.append({
+                '_fossil': 'categoryMetadata',
+                'type': 'Category',
+                'name': category.title,
+                'id': category.id,
```

Comment:
Yes but when adapting the code for events/timetable, we didn't convert them. Check for `'id': ` in the file.